app/services/record_participant_event.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class RecordParticipantEvent
+  include InitializeWithConfig
+  class << self
+    def required_params
+      %i[participant_id lead_provider declaration_type declaration_date raw_event]
+    end
+  end
+
+  def call
+    (set_config_ect_profile || :not_found) &&
+      (create_record || :unprocessable_entity) &&
+      (invalid_provider || :unprocessable_entity) &&
+      :no_content
+  end
```

Comment:
What am I looking at here? If we have no profile, what exactly is going to happen?

Can we make use of some early returns here? And explicit if statements?

Reply:
Are you having problems associating this logical flow without branches?
is:

```
return :not_found unless set_config_ect_profile
return :unprocessable_entity unless create_record
return :unprocessable_entity unless invalid_provider
:no_content`

```
really clearer?

(I deliberately added in the `(invalid_provider || :unprocessable_entity) &&` as a separate line in a live session with Matt to make sure that we could map the required return states to the condition that generated the error.

Reply:
I find it much easier to read the second version.

In the second version, is the order correct? Should we check for the invalid provider before we try to create a record?

Reply:
also
```
return :unprocessable_entity unless invalid_provider
```
reads a bit surprising - I would expect
```
return :unprocessable_entity if invalid_provider
```
Cause if the provider is not invalid, then why would the entity not be processable?

Reply:
> In the second version, is the order correct? Should we check for the invalid provider before we try to create a record?

Yes. The logic was that we would want to create the record anyway, but flag that it was for a different provider.

Reply:
`invalid_provider` was incorrect. Fixed.